packages/grid/_modules_/grid/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -0,0 +1,252 @@
+import * as React from 'react';
+import ownerDocument from '@material-ui/core/utils/ownerDocument';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridApiMethod } from '../../root/useGridApiMethod';
+import { GridPrintExportApi } from '../../../models/api/gridPrintExportApi';
+import { useLogger } from '../../utils/useLogger';
+import { useGridSelector } from '../core/useGridSelector';
+import { visibleSortedGridRowsSelector } from '../filter/gridFilterSelector';
+import { useGridState } from '../core/useGridState';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { GridPrintExportOptions } from '../../../models/gridExport';
+import { allGridColumnsSelector } from '../columns/gridColumnsSelector';
+
+type PrintWindowOnLoad = (
+  printWindow: HTMLIFrameElement,
+  options?: Pick<
+    GridPrintExportOptions,
+    'copyStyles' | 'bodyClass' | 'pageStyle' | 'hideToolbar' | 'hideFooter'
+  >,
+) => void;
+
+export const useGridPrintExport = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'rowHeight'>,
+): void => {
+  const logger = useLogger('useGridPrintExport');
+  const [, setGridState, forceUpdate] = useGridState(apiRef);
+  const visibleSortedRows = useGridSelector(apiRef, visibleSortedGridRowsSelector);
+  const columns = useGridSelector(apiRef, allGridColumnsSelector);
+  const doc = React.useRef<Document | null>(null);
+  const previousGridState = React.useRef<any>(null);
+  const previousHiddenColumns = React.useRef<string[]>([]);
+
+  React.useEffect(() => {
+    doc.current = ownerDocument(apiRef!.current.rootElementRef!.current as HTMLElement);
+  }, [apiRef]);
+
+  const prepGridForPrint = React.useCallback(
+    (fields?: string[], allColumns?: boolean): void => {
+      // Show only wanted columns
+      if (fields && fields.length) {
+        apiRef!.current.updateColumns(
+          columns.map((column) => {
+            if (!column.hide) {
+              previousHiddenColumns.current.push(column.field);
+            }
+
+            // Show all columns
+            if (allColumns) {
+              column.hide = true;
+              return column;
+            }
+
+            column.hide = !fields.includes(column.field);
+
+            return column;
+          }),
+        );
+      }
+
+      setGridState((state) => {
+        previousGridState.current = state;
+        return {
+          ...state,
+          containerSizes: {
+            ...state.containerSizes!,
+            renderingZone: {
+              ...state.containerSizes!.renderingZone,
+              height: visibleSortedRows.size * props.rowHeight!,
+            },
+            renderingZonePageSize: visibleSortedRows.size,
+          },
+          viewportSizes: {
+            ...state.viewportSizes,
+            height: visibleSortedRows.size * props.rowHeight!,
+          },
+        };
+      });
```

Comment:
Ok, I took a bit more time to check this more in-depth. The problem is that this logic doesn't really do enable `autoHeight` neither it disables the virtualization. What it does is force the grid to render all rows (have all the HTML for the rows be in the dom but still keep the scroll so that the content on the page doesn't jump). 

For example @m4theushw suggestion to reuse his logic from the `useGridNoVirtualization` won't work because as far as can see there are other places where that prop is directly red and used (same as the `autoHeight` , which was my original idea). I'll see if moving that chunk into `useGridNoVirtualization` will make sense and expose it as an API from there.